### DISCUSSION TITLE
STARSolo settings for SureCell

### DISCUSSION THREAD

**Discussion by varemo** [alexdobin/STAR#1258] [Q&A]
Hi, I am trying to figure out if I can use STARSolo to process SureCell data. 

R1 starts with a phase block (sequence of variable length across different reads) followed by 3 barcodes separated by 2 linkers, followed by ACG and then finally the UMI. See page 2 of [this pdf](https://www.illumina.com/content/dam/illumina-marketing/documents/products/technotes/single-cell-rna-algorithm-tech-note-1070-2016-015.pdf) from Illumina for clearer illustration and suggested method to decode R1.

My guess is that I can use `soloType CB_UMI_Complex` and three sets of strings for `soloCBposition` that use the respective linkers as anchors. But I am not sure if `soloAdapterSequence` accepts more than one sequence to be used as anchor?

Any pointers as how to piece this together?

Many thanks
/Leif

**Comment by alexdobin** [comment]
Hi Leif,

multiple adapters are not presently supported - however, you can just pick one of the adapters (Linker 1 or 2) and record the barcodes and UMI coordinates with respect to this adapter.
Please let me know whether it works for you, I can help with coordinate selection if there are any issues.

Cheers
Alex

**Reply by varemo** [reply]
Thanks Alex! 

I think what initially mislead me was the phrasing for the `CB_UMI_Complex` description: 

"one UMI of fixed length, but multiple Cell Barcodes of varying length, as well as adapters sequences are allowed in read2 only, e.g. inDrop."

I thought "multiple" applied to adapters as well :)

Anyway, you are right, I'll use one of the linkers as anchors. 

Did I get it correct, that I should have 3 whitelist files, one per barcode position, or will the barcodes be concatenated to one long barcode which will be matched to 1 whitelist file with possible concatenated barcode combinations?

Thanks again for all the great work you put into this analysis tool!

Cheers
/Leif

**Reply by alexdobin** [reply]
Hi Leif,

you are right, that language was confusing, I fixed it.

Indeed you need 3 whitelist files for each barcode position, they are not concatenated. The files should be given in the same order you provide their coordinates in ```--CB_UMI_Complex```. Also, the coordinates are 0-based!

Cheers
Alex

**Reply by varemo** [reply]
If anyone returns to this thread, here is how I ended up doing this:

```
--soloType CB_UMI_Complex
--soloCBposition 2_-6_2_-1 3_1_3_6 3_22_3_27
--soloUMIposition 3_31_3_38
--soloAdapterSequence TAGCCATCGCATTGC
```

Here is the barcode/umi configuration for reference (copied from the pdf linked in the original post):
![image](https://user-images.githubusercontent.com/12655718/122553334-d62d3e00-d037-11eb-8976-7128102913f2.png)

In addition I supplied the same barcode file 3 times (as the same pool of 96 barcodes are recycled for all the 3 barcode positions):
```
--soloCBwhitelist barcodes.txt barcodes.txt barcodes.txt
```
@alexdobin Is this needed or would STAR use the same file for all barcode position if only specified once?

Would also be great if someone can verify that I got the `--solo[CB|UMI]position` arguments correctly!

Thanks!

**Reply by alexdobin** [reply]
Hi Leif,

many thanks for sharing your setup!
Yes, you need to specify the file 3 times, even though it's the same file.
The best way to check whether your settings are correct is to map the real data and see what proportion of reads get the barcodes recovered. You can check some useful statistics in the Solo.out/Barcodes.stats file.

Cheers
Alex
